panels/led.py

```diff
@@ -0,0 +1,169 @@
+import gi
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk, GLib
+from ks_includes.KlippyGcodes import KlippyGcodes
+from ks_includes.screen_panel import ScreenPanel
+
+RGB = ["r", "g", "b"]
+RGBW = ["r", "g", "b", "w"]
+RGB_PRESETS = {"r": [1.0, 0.0, 0.0],
+               "g": [0.0, 1.0, 0.0],
+               "b": [0.0, 0.0, 1.0],
+               "rgb": [1.0, 1.0, 1.0],
+               "off": [0.0, 0.0, 0.0]}
+RGBW_PRESETS = {"r": [1.0, 0.0, 0.0, 0.0],
+                "g": [0.0, 1.0, 0.0, 0.0],
+                "b": [0.0, 0.0, 1.0, 0.0],
+                "w": [0.0, 0.0, 0.0, 1.0],
+                "rgb": [1.0, 1.0, 1.0, 0.0],
+                "rgbw": [1.0, 1.0, 1.0, 1.0],
+                "off": [0.0, 0.0, 0.0, 0.0]}
+
+
+class Panel(ScreenPanel):
+
+    def __init__(self, screen, title):
+        super().__init__(screen, title)
+        self.col_mix = RGBW
+        self.presets = {}
+        self.scales = {}
+        self.buttons = []
+        self.leds = self._printer.get_leds()
+        self.led = self.leds[0] if len(self.leds) == 1 else None
+        self.open_selector(None, self.led)
+
+    def activate(self):
+        if self.led is not None:
+            self.set_title(f"{self.led}")
+
+    def set_title(self, title):
+        self._screen.base_panel.set_title(self.prettify(title))
+
+    def back(self):
+        if len(self.leds) > 1:
+            self.set_title(self._screen.panels[self._screen._cur_panels[-1]].title)
+            self.open_selector(led=None)
+            return True
+        return False
+
+    def open_selector(self, widget=None, led=None):
+        for child in self.content.get_children():
+            self.content.remove(child)
+        if led is None:
+            self.content.add(self.led_selector())
+        else:
+            self.content.add(self.color_selector(led))
+        self.content.show_all()
+
+    def led_selector(self):
+        self.led = None
+        columns = 3 if self._screen.vertical_mode else 4
+        grid = self._gtk.HomogeneousGrid()
+        for i, led in enumerate(self.leds):
+            name = led.split()[1] if len(led.split()) > 1 else led
+            button = self._gtk.Button(None, name.upper(), style=f"color{(i % 4) + 1}")
+            button.connect("clicked", self.open_selector, led)
+            grid.attach(button, (i % columns), int(i / columns), 1, 1)
+        scroll = self._gtk.ScrolledWindow()
+        scroll.add(grid)
+        return scroll
+
+    def color_selector(self, led):
+        logging.info(led)
+        self.led = led
+        self.set_title(f"{self.led}")
+        grid = self._gtk.HomogeneousGrid()
+        color_data = self._printer.get_led_color(led)
+        color_mix = self._printer.get_led_color_mix(led)
+        if color_data is None or color_mix is None:
+            self.back()
+            return
+        if len(color_mix) == 3:
+            self.col_mix = RGB
+            if len(color_data) > 3:
+                # is it always rgbw?
```

Comment:
From my experience, most led modules/led strips are 3 base colors [+ W warm/cold/light temp, K]

Reply:
it has nothing to do with the physical strips, the note was left as a reminder to check if klipper always uses 4 channels and the answer is yes
https://www.klipper3d.org/Status_Reference.html#led